WatchFaceKotlin/wear/src/main/java/com/example/android/wearable/watchfacekotlin/config/AnalogComplicationConfigRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,754 @@
+import android.util.Log
+import android.view.Gravity
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.Switch
+import android.widget.Toast
+
+import androidx.core.content.edit
+import androidx.recyclerview.widget.RecyclerView
+
+import com.example.android.wearable.watchfacekotlin.R
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.BackgroundComplicationConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.ColorConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.ConfigItemType
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.MoreOptionsConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.PreviewAndComplicationsConfigItem
+import com.example.android.wearable.watchfacekotlin.model.AnalogComplicationConfigData.UnreadNotificationConfigItem
+import com.example.android.wearable.watchfacekotlin.watchface.AnalogComplicationWatchFaceService
+
+import java.util.concurrent.Executors
+
+/**
+ * Displays different layouts for configuring watch face's complications and appearance settings
+ * (highlight color [second arm], background color, unread notifications, etc.).
+ *
+ *
+ * All appearance settings are saved via [SharedPreferences].
+ *
+ *
+ * Layouts provided by this adapter are split into 5 main view types.
+ *
+ *
+ * A watch face preview including complications. Allows user to tap on the complications to
+ * change the complication data and see a live preview of the watch face.
+ *
+ *
+ * Simple arrow to indicate there are more options below the fold.
+ *
+ *
+ * Color configuration options for both highlight (seconds hand) and background color.
+ *
+ *
+ * Toggle for unread notifications.
+ *
+ *
+ * Background image complication configuration for changing background image of watch face.
+ */
+class AnalogComplicationConfigRecyclerViewAdapter(
+    private val context: Context,
+    watchFaceServiceClass: Class<*>,
+    settingsDataSet: ArrayList<ConfigItemType>
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+    /**
+     * Used by associated watch face ([AnalogComplicationWatchFaceService]) to let this
+     * adapter know which complication locations are supported, their ids, and supported
+     * complication data types.
+     */
+    enum class ComplicationLocation {
+        BACKGROUND, LEFT, RIGHT, TOP, BOTTOM
+    }
+
+    // ComponentName associated with watch face service (service that renders watch face). Used
+    // to retrieve complication information.
+    private val watchFaceComponentName: ComponentName
+    private val settingsDataSet: ArrayList<ConfigItemType>
+    var sharedPref: SharedPreferences
```

Comment:
Don't work with SharedPreferences directly in the adapter. This is the kind of work that should happen in a ViewModel